component/api/src/main/java/org/exoplatform/social/core/storage/api/SpaceStorage.java

```diff
@@ -717,6 +717,37 @@ default List<String> getMemberSpaceIds(String identityId, int offset, int limit)
     return getMemberRoleSpaceIdentityIds(identityId, offset, limit);
   }
 
+  /**
+   * Retrieves the list of Space {@link Identity} technical identifiers
+   *
+   * @param identityId user {@link Identity} technical identifier
+   * @param status {@link String} equals to MEMBER, MANAGER, REDACTOR, PENDING,
+   *          INVITED, IGNORED
+   * @param offset The starting point
+   * @param limit The limitation of returned results
+   * @return {@link List} of space {@link Identity} technical identifiers
+   * @throws SpaceStorageException
+   */
+  List<String> getSpaceIdentityIdsByUserRole(String identityId,
+                                             String status,
+                                             int offset,
+                                             int limit) throws SpaceStorageException;
+
+  /**
+   * Retrieves the list of favorite Space {@link Identity} technical identifiers
+   *
+   * @param identityId user {@link Identity} technical identifier
+   * @param spaceFilter
+   * @param offset The starting point
+   * @param limit The limitation of returned results
+   * @return {@link List} of space {@link Identity} technical identifiers
+   * @throws SpaceStorageException
+   */
+  List<String> getFavoriteSpaceIdentityIds(String identityId,
+                                           SpaceFilter spaceFilter,
+                                           int offset,
+                                           int limit) throws SpaceStorageException;
```

Comment:
```suggestion
                                           int limit);
```
useless unchecked exception